File: Park/day_1/ParkProblem/ParkProblem.Test/UnitTest1.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace ParkProblem.Test
{
    [TestClass]
    public class UnitTest1
    {
        MathMachine mathMachine;

        [TestInitialize]
        public void setUp()
        {
            mathMachine = new MathMachine();
        }

        [TestMethod]
        public void 데이터_없음_조합_없음()
        {
            int[] nums = {};

            string expected = "{}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_하나_조합_하나()
        {
            int[] nums = { 7 };

            string expected = "{7}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_하나_조합_없음()
        {
            int[] nums = { 6 };

            string expected = "{}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_하나_조합_없음_v1()
        {
            int[] nums = { 1 };

            string expected = "{1,1}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 2));
        }

        [TestMethod]
        public void 데이터_둘_조합_없음()
        {
            int[] nums = { 5, 6 };

            string expected = "{}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_둘_조합_하나()
        {
            int[] nums = { 1, 6 };

            string expected = "{1,6}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_둘_조합_둘_v2()
        {
            int[] nums = { 7, 0 };

            string expected = "{7}, {7,0}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_둘_조합_둘_v3()
        {
            int[] nums = { 0, 7 };

            string expected = "{7}, {0,7}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_둘_조합_둘_v4()
        {
            int[] nums = { 3, 4 };

            string expected = "{3,3}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 6));
        }

        [TestMethod]
        public void 데이터_둘_조합_둘_v5()
        {
            int[] nums = { 7, 7 };

            string expected = "{7}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 7));
        }

        [TestMethod]
        public void 데이터_둘_조합_둘_v6()
        {
            int[] nums = { 4, 3 };

            string expected = "{3,3}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 6));
        }

        [TestMethod]
        public void 데이터_둘_조합_둘_v7()
        {
            int[] nums = { 3, 1 };

            string expected = "{1,1}";

            Assert.AreEqual(expected, mathMachine.numOfCombination(nums, 2));
        }



    }
}

File: MaxBinaryTreeCpp/MaxBinaryTreeCpp/MaxBinaryTreeCppTests/unittest1.cpp
#include "stdafx.h"
#include "CppUnitTest.h"

#include <vector>
#include "..\MaxBinaryTreeCpp\TreeNode.h"
#include "..\MaxBinaryTreeCpp\BinaryTreeConstructor.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace std;

namespace MaxBinaryTreeCppTests
{		
	TEST_CLASS(UnitTest1)
	{
		BinaryTreeConstructor *constructor;

	public:
		TEST_METHOD_INITIALIZE(SetUp)
		{
			constructor = new BinaryTreeConstructor();
		}

		TEST_METHOD_CLEANUP(CleanUp)
		{
			delete constructor;
		}

		TEST_METHOD(New_TreeNode)
		{
			TreeNode *node = new TreeNode(1);

			Assert::AreEqual(1, node->val);
			delete node;
		}

		TEST_METHOD(NullArray)
		{
			vector<int> nums = { };
			TreeNode *root = constructor->constructMaximumBinaryTree(nums);
			Assert::IsNull(root);
		}

		TEST_METHOD(OneArray)
		{
			vector<int> nums = { 1 };
			TreeNode *root = constructor->constructMaximumBinaryTree(nums);
			Assert::AreEqual(1, root->val);
		}

		TEST_METHOD(TwoArray_right)
		{
			vector<int> nums = { 2, 1 };
			TreeNode *root = constructor->constructMaximumBinaryTree(nums);
			Assert::AreEqual(2, root->val);
			Assert::IsNull(root->left);
			Assert::AreEqual(1, root->right->val);
		}

		TEST_METHOD(TwoArray_left)
		{
			vector<int> nums = { 1, 2 };
			TreeNode *root = constructor->constructMaximumBinaryTree(nums);
			Assert::AreEqual(2, root->val);
			Assert::AreEqual(1, root->left->val);
			Assert::IsNull(root->right);
		}

		TEST_METHOD(Final)
		{
			vector<int> nums = { 3, 2, 1, 6, 0, 5 };

			TreeNode *root = constructor->constructMaximumBinaryTree(nums);
			Assert::AreEqual(6,		root->val);
			Assert::AreEqual(3,		root->left->val);
			Assert::AreEqual(5,		root->right->val);
			Assert::IsNull(			root->left->left);
			Assert::AreEqual(2,		root->left->right->val);
			Assert::AreEqual(0,		root->right->left->val);
			Assert::IsNull(			root->right->right);
			Assert::IsNull(			root->left->right->left);
			Assert::AreEqual(1,		root->left->right->right->val);
		}
	};
}